compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1630,6 +1630,10 @@ public void visit(BLangBinaryExpr binaryExpr) {
     }
 
     private boolean validateBinaryExpr(BLangBinaryExpr binaryExpr) {
+        // This is when there are semantic errors causing lsh type to be null.
+        if (binaryExpr.lhsExpr.type == null) {
```

Comment:
Not related to this PR I guess, but shouldn't the LHS type be set to `BSemanticErrorType` if there were semantic errors in `lhsExpr`?

Reply:
This check is removed as the bug causing this is fixed.